test/test_cartesian_dmp.py

```diff
@@ -31,6 +32,18 @@ def test_imitate_cartesian_dmp():
     assert_array_almost_equal(Y, Y2, decimal=2)
 
 
+def test_imitate_cartesian_dmp_rounding_error():
+    dt = 0.01
+    execution_time = 6.0
+
+    dmp = CartesianDMP(
+        execution_time=execution_time, dt=dt,
+        n_weights_per_dim=10)
+    T, Y = dmp.open_loop()
+    assert_equal(len(T), 601)
+    assert_equal(len(Y), 601)
```

Comment:
I don't think that's the behavior one would expect (6s/0.01s = 601 steps), let's assert equal to 600 and call it a bug. For example, a user comparing a 6s DMP to six concatenated 1s DMPs wouldn't know where to remove the extra points in order to plot them together.

The rounding error can be prevented by introducing a Decimal type for dt. If this solution is not possible, let's offer a flag to select from unreliable step lengths (linspace) or an unreliable number of steps (current behavior).

Or are you sure you want to keep it 601? Then, code is fine and ready to merge. :-)

Reply:
Yes, it should be 601 in this case because we include step 0. However, it was 602 steps previously.

Reply:
Right, that was step 0, thanks for the reminder.

Numpy's arange is definitely an improvement. `np.arange(0, 6.01, 0.01)[-1]` is 6.0, whereas the iterator approach fails as discovered.

But I'm not sure if I'd choose it here considering that it still "...may result in the last element of out being greater than stop" (a filter for timesteps would discard the last point), it has "precision loss ... when start is much larger than step" (possibly an issue for imitation learning when the recording is long) and then it is not clear if the mentioned "unexpected behaviour" is only int32-related. 

Decimal types seem safer for time variables, however arange on floats fixes the current issue and requires only small code changes, so I agree.

Reply:
So I'll merge this. Maybe we can find a better solution in the future.